src/main/ruby/truffleruby/core/file.rb

```diff
@@ -1161,13 +1164,12 @@ class << self
   def initialize(path_or_fd, mode = nil, perm = nil, **options)
     if Primitive.is_a?(path_or_fd, Integer)
       super(path_or_fd, mode, **options)
-      @path = nil
     else
       path = Truffle::Type.coerce_to_path path_or_fd
       nmode, _binary, _external, _internal, _autoclose, perm = Truffle::IOOperations.normalize_options(mode, perm, options)
       fd = IO.sysopen(path, nmode, perm)
 
-      @path = path
+      options[:path] = path
       super(fd, mode, **options)
```

Comment:
```suggestion
      super(fd, mode, path: path, **options)
```

Reply:
Hmm, I think this fails to override the passed value of :path if the user happens to pass it to ` File#new`. I tried a small experiment :

```
irb(main):001:1* def x(**options)
irb(main):002:1*   puts options
irb(main):003:0> end
=> :x
irb(main):004:1* def xoverride(**options)
irb(main):005:1*   x(overriden: "overriden", **options)
irb(main):006:0> end
=> :xoverride
irb(main):007:0> xoverride(option1: "option1", overriden: "no")
{:overriden=>"no", :option1=>"option1"}
=> nil      
```                                            
So it's indeed the passed argument that "wins". Whereas if you try : 

```
irb(main):014:0> File.new("/home/mostafa/test.rb", path: "invalid").path
=> "/home/mostafa/test.rb"
```
The desired behaviour is that the passed argument is ignored.

Reply:
How about `super(fd, mode, **options, path: path)` then?

Reply:
Works.